container/build_tar.py

```diff
@@ -284,6 +290,34 @@ def add_pkg_metadata(self, metadata_tar, deb):
       raise self.DebError('Unknown Exception {0}. Please report an issue at'
                           ' github.com/bazelbuild/rules_docker.'.format(e))
 
+  def add_rpm(self, rpm):
+    """Extract an rpm package in the output tar.
+
+    All files presents in that rpm package will be added to the
+    output tar under the same paths. No user name nor group names will
+    be added to the output.
+
+    Args:
+      rpm: the tar file to add
+
+    Raises:
+      RPMError: if the format of the rpm archive is incorrect.
+    """
+    try:
+      p1 = subprocess.Popen(["rpm2cpio", rpm], stdout=subprocess.PIPE)
```

Comment:
I do not sure if it good to assume that we had `rpm2cpio` by default.

Reply:
Yes. I think that we would need a pure python solution. My main issue was that I could find all the needed pieces written in pure python, but not in a reasonable shape or one place, so that I could just rely on one or more pip dependencies.

Reply:
pure-python is a bit of a wormhole i think. and if there is `rpm` package installed, it provides `rpm2cpio`